src/config-wrapper.js

```diff
@@ -0,0 +1,143 @@
+      if (data.get(name)) {
+        // eslint-disable-next-line no-param-reassign
+        prev[name] = Number.isNaN(Number(data.get(name)))
+          ? data.get(name)
+          : Number.parseInt(data.get(name), 10);
+      }
+      return prev;
+    }, {});
+  } else if (/json/.test(request.headers.get('content-type'))) {
+    const data = await request.json();
+    const ret = names.reduce((prev, name) => {
+      if (data[name]) {
+      // eslint-disable-next-line no-param-reassign
+        prev[name] = data[name];
+      }
+      return prev;
+    }, {});
+    return ret;
+  }
+  // just return the request parameters
+  const data = new URLSearchParams(new URL(request.url).search);
+  return names.reduce((prev, name) => {
+    if (data.get(name)) {
+      // eslint-disable-next-line no-param-reassign
+      prev[name] = Number.isNaN(Number(data.get(name)))
+        ? data.get(name)
+        : Number.parseInt(data.get(name), 10);
+    }
+    return prev;
+  }, {});
+}
+
+function getToken({ headers }) {
+  if (headers.get('authorization')) {
+    return headers.get('authorization');
+  }
+  if (headers.get('x-github-token')) {
+    return `token ${headers.get('x-github-token')}`;
+  }
+  return undefined;
+}
+
+function wrap(func, required, ...configs) {
+  return async (request, context) => {
+    const { owner, repo, ref } = await getData(request, 'owner', 'repo', 'ref');
+    const {
+      transactionId,
+    } = context.invocation;
+
+    const options = {
+      headers: {
+        authorization: getToken(request),
+      },
+    };
+
+    // init is a helper function in helix-fetch that makes it easy
+    // to recreate a request by returning the inital options
+    const newreq = new Request(request.url, request.init);
+
+    context.config = {};
+    if (!!owner && !!repo && !!ref) {
+      try {
+        const config = await configs
+          .filter((name) => !!loaders[name])
+          .reduce(async (confp, name) => {
+            const conf = await confp;
+            try {
+              conf[name] = await new loaders[name]()
+                .withTransactionID(transactionId)
+                .withRepo(owner, repo, ref, options)
+                .withLogger(context.log)
+                .init();
+
+              if (!conf[name].source) {
+                throw new Error(`${name} config is empty`);
+              }
+            } catch (e) {
+              if (required) {
+                throw new Error(`Unable to load ${name} config for ${owner}, ${repo}, ${ref}: ${e.message}`);
+              } else if (context.log) {
+                context.log.warn(`Unable to load ${name} config for ${owner}, ${repo}, ${ref}: ${e.message}`);
+              }
+            }
+            return conf;
+          }, {});
+
+        context.config = config;
+      } catch (e) {
+        return new Response(e.message, {
+          status: 502,
+          headers: {
+            'Content-Type': 'text/plain',
+            'x-error': e.message,
```

Comment:
clean headers?

Reply:
Do you remember where we first added the header cleaning function?